pipenv/utils/resolver.py

```diff
@@ -646,6 +707,8 @@ def constraints(self):
                 )
                 for c in self.parsed_constraints
             ]
+            if self.dev:
```

Comment:
Could you explain more why we only include the default_constraints when dev is specified?  I don't think I quite understand out of the gate.  I got to looking at this more because this is the only place that uses self.dev, and I am noticing how many parameters `pip_install` takes already.

Reply:
I think constraint is used only when installing dev-packages. If it does with new normal packages, the installing package could not overwrite existing packages, because it use them as contraint, and therefore might not be installed.

Reply:
Ah ok thanks, that makes sense.